### PR TITLE
IntegrationTest WIP

### PR DESCRIPTION
**Pull request by markrmiller**

### CODE REVIEW

**Comment by markrmiller** [comment]
I've got to take a look at why there are conflicting files, kind of surprised on that, but should be simple to resolve.

This has a variety of things required to test from an update going to the producer to the consumer to Solr.

The test is bare bones and needs some better logic than the waits it currently does, but will be quick and simple, just have been waiting to solve this issue I have where the update ends up at Solr with no data in the Solr request.

Not sure if it's due to the serializer or deserializer method, but the request ends up with no docs in it. Soon as that is resolved I'll clean up the test and put in some validation checks.

File: crossdc-consumer/src/main/java/org/apache/solr/crossdc/common/MirroredSolrRequestSerializer.java [hunk 1]
```diff
@@ -16,16 +16,23 @@
...
+import org.slf4j.LoggerFactory;
 
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.lang.invoke.MethodHandles;
 import java.util.Map;
 
-public class MirroredSolrRequestSerializer implements Serializer<MirroredSolrRequest> {
+public class MirroredSolrRequestSerializer implements Serializer<MirroredSolrRequest>, Deserializer<MirroredSolrRequest> {
```
Comment (anshumg) [inline crossdc-consumer/src/main/java/org/apache/solr/crossdc/common/MirroredSolrRequestSerializer.java:33]: I think we should just split this into a serializer and deserializer.

**Review by anshumg — COMMENTED** [review]
Some conflicts came up after I merged my other (producer wip) PR. I'll take a look at it.

**Comment by anshumg** [comment]
Merging this in so we're working off the same base.

**Pull request merged**